run.py

```diff
@@ -100,29 +100,42 @@ def get_acq_or_file_path(container) -> str:
 
 def get_hdr_fields(dicom: FileListOutput, site: str) -> dict:
     """Get relevant fields from dicom header of an acquisition"""
+    # Reload the dicom file to ensure dicom header is loaded
+    dicom = dicom.reload()
+    dcm = get_acq_or_file_path(dicom)
 
     if "file-classifier" not in dicom.tags or "header" not in dicom.info:
-        log.error(
-            f"File-classifier gear has not been run on {get_acq_or_file_path(dicom)}"
-        )
+        log.error("File-classifier gear has not been run on %s", dcm)
         return {"error": "FILE_CLASSIFIER_NOT_RUN"}
 
-    dcm_hdr = dicom.reload().info["header"]["dicom"]
-    hdr_fields = {"error": None, "site": site, "ses_id": dicom.parents.session}
+    dcm_hdr = dicom.info["header"]["dicom"]
+    meta = {"error": None, "site": site, "ses_id": dicom.parents.session}
 
-    for key, function in DICOM_FUNCTION_DICT.items():
-        try:
-            hdr_fields[key] = function(dcm_hdr, site)
-        except (KeyError, ValueError, AttributeError):
-            hdr_fields[key] = None
-            log.error(f"{get_acq_or_file_path(dicom)} is missing {key}.")
-            hdr_fields["error"] = "MISSING_DICOM_FIELDS"
-        except ValueError:
-            hdr_fields[key] = None
-            log.error(f"{key} in {get_acq_or_file_path(dicom)} failed to parse.")
-            hdr_fields["error"] = "DICOM_FIELDS_PARSE_ERROR"
+    try:
+        meta["pi_id"] = parse_dicom_hdr.parse_pi(dcm_hdr, site).casefold()
+    except (KeyError, ValueError):
+        log.debug("%s problem fetching PI ID", dcm)
+        meta["error"] = "MISSING_DICOM_FIELDS"
 
-    return hdr_fields
+    try:
+        meta["sub_id"] = parse_dicom_hdr.parse_sub(dcm_hdr, site).casefold()
+    except KeyError:
+        log.debug("%s problem fetching SUB ID", dcm)
+        meta["error"] = "MISSING_DICOM_FIELDS"
```

Comment:
is the `AttributeError` catch no longer needed here (and on downstream checks?)

Reply:
I'm not 100% sure, but I don't see a situation in which that would arise, and also haven't seen them occur in the redcap gear which doesn't catch them. Do you a see a reason to keep it?